back/spec/models/notifications/project_phase_started_spec.rb

```diff
@@ -8,12 +8,24 @@
       phase = create(:phase)
       activity = create(:activity, item: phase, action: 'started')
 
-      notifications = Notifications::ProjectPhaseStarted.make_notifications_on activity
+      notifications = described_class.make_notifications_on activity
       expect(notifications.first).to have_attributes(
         recipient_id: user.id,
         phase_id: phase.id,
         project_id: phase.project_id
       )
     end
+
+    it 'only creates a notification for users who can participate' do
+      phase = create :phase
+      activity = create :activity, item: phase, action: 'started'
+      create_list :user, 5
+      allow_any_instance_of(ParticipationContextService).to(
+        receive(:participation_possible_for_context?).and_return(true, false, true, false)
```

Comment:
Without looking at the code, it is not clear for the reader why `and_return` takes 4 values. Why will `participation_possible_for_context?` be called 4 times (or more, because the last value of `and_return` will be returned when more than 4 calls are done)? Wouldn't it be enough to have 2 users, with only 1 who participates? Why do we need 5 in this test?

Reply:
No indeed, I was having some trouble stubbing the `participation_possible_for_context?`. What I had in mind is to replace it by a simple implementation that would look at the user ID for example and return true or false based on that. Then, instead of verifying the size of the notifications, I would have matched the recipient IDs from the notifications against the user IDs for which the stubbed method would return true. I think the test would be more precise that way.

Do you think it's fine to keep the test as it's written now and reduce the users to two? In that case, I'll make the change.

Reply:
Yes, that is fine.